File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/KeyForDidTest.cs
using Hyperledger.Indy.LedgerApi;
using Hyperledger.Indy.DidApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{

    [TestClass]
    public class KeyForDidTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestKeyForDidWorksForMyDid()
        {
            var result = await Did.CreateAndStoreMyDidAsync(wallet, "{}");
            var did = result.Did;
            var key = result.VerKey;

            var receivedKey = await Did.KeyForDidAsync(pool, wallet, did);

            Assert.AreEqual(key, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForTheirDid()
        {
            var identityJson = string.Format(IDENTITY_JSON_TEMPLATE, DID_MY1, VERKEY_MY1);
            await Did.StoreTheirDidAsync(wallet, identityJson);

            var receivedKey = await Did.KeyForDidAsync(pool, wallet, DID_MY1);

            Assert.AreEqual(VERKEY_MY1, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForGetKeyFromLedger()
        {
            var result = await Did.CreateAndStoreMyDidAsync(wallet, TRUSTEE_IDENTITY_JSON);
            var trusteeDid = result.Did;

            var identityJson = string.Format(IDENTITY_JSON_TEMPLATE, DID_MY1, VERKEY_MY1);
            await Did.StoreTheirDidAsync(wallet, identityJson);

            var nymRequest = await Ledger.BuildNymRequestAsync(trusteeDid, DID_MY1, VERKEY_MY1, null, null);
            await Ledger.SignAndSubmitRequestAsync(pool, wallet, trusteeDid, nymRequest);

            var receivedKey = await Did.KeyForDidAsync(pool, wallet, DID_MY1);

            Assert.AreEqual(VERKEY_MY1, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForNoKey()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStateException>(() =>
               Did.KeyForDidAsync(pool, wallet, DID_MY2)
           );
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/did/KeyForDidTest.java
package org.hyperledger.indy.sdk.did;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.ledger.Ledger;
import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class KeyForDidTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	@Test
	public void testKeyForDidWorksForMyDid() throws Exception {
		DidResults.CreateAndStoreMyDidResult result = Did.createAndStoreMyDid(wallet, "{}").get();
		String did = result.getDid();
		String key = result.getVerkey();

		String receivedKey = Did.keyForDid(pool, wallet, did).get();
		assertEquals(key, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForTheirDid() throws Exception {
		Did.storeTheirDid(wallet, String.format(IDENTITY_JSON_TEMPLATE, DID_MY1, VERKEY_MY1)).get();

		String receivedKey = Did.keyForDid(pool, wallet, DID_MY1).get();
		assertEquals(VERKEY_MY1, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForGetKeyFromLedger() throws Exception {
		DidResults.CreateAndStoreMyDidResult result = Did.createAndStoreMyDid(this.wallet, TRUSTEE_IDENTITY_JSON).get();
		String trusteeDid = result.getDid();

		Did.storeTheirDid(wallet, String.format(IDENTITY_JSON_TEMPLATE,  DID_MY1, VERKEY_MY1)).get();
		String nymRequest = Ledger.buildNymRequest(trusteeDid,  DID_MY1, VERKEY_MY1, null, null).get();
		Ledger.signAndSubmitRequest(pool, wallet, trusteeDid, nymRequest).get();

		String receivedKey = Did.keyForDid(pool, wallet, DID_MY1).get();
		assertEquals(VERKEY_MY1, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForNoKey() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		Did.keyForDid(pool, wallet, DID_MY2).get();
	}
}